components/automate-ui/src/app/components/resource-dropdown/resource-dropdown.component.spec.ts

```diff
@@ -30,133 +32,356 @@ describe('ResourceDropdownComponent', () => {
           genResource('other resource3', false),
           genResource('other resource4', false)
-        ]],
+        )]],
       ['multiple resources but just one checked at the bottom',
-        [
+        [genResourceList(
           genResource('other resource1', false),
           genResource('other resource2', false),
           genResource('target resource', true)
-        ]]
-    ], function (description: string, resources: ResourceChecked[]) {
+        )]]
+    ], function (description: string, resources: ResourceCheckedSection[]) {
       it(`displays resource name for ${description}`, () => {
-        component.resources = resources;
-        component.ngOnChanges(
-          { resources: new SimpleChange([], resources, true) });
+        changeResources([], resources, true);
         expect(component.label).toEqual('target resource');
       });
     });
 
     using([
       ['some checked and some not checked', 3,
-        [
+        [genResourceList(
           genResource('name1', true),
           genResource('name2', false),
           genResource('name3', true),
           genResource('name4', true),
           genResource('name5', false)
-        ]],
+        )]],
       ['all checked', 4,
-        [
+        [genResourceList(
           genResource('name1', true),
           genResource('name2', true),
           genResource('name3', true),
           genResource('name4', true)
-        ]],
+        )]],
       ['just two and both checked', 2,
-        [
+        [genResourceList(
           genResource('name1', true),
           genResource('name2', true)
-        ]]
-    ], function (description: string, count: number, resources: ResourceChecked[]) {
+        )]]
+    ], function (description: string, count: number, resources: ResourceCheckedSection[]) {
       it(`displays resource designator with count for multiple resources with ${description}`,
         () => {
-          component.resources = resources;
           component.objectNounPlural = 'lemurs';
-          component.ngOnChanges(
-            { resources: new SimpleChange([], resources, true) });
+          changeResources([], resources, true);
           expect(component.label).toEqual(`${count} lemurs`);
         });
     });
   });
 
+  function toggleDropdown() {
+    // The event from an actual click on the dropdown fires *both* of these in the order given.
+    component.toggleDropdown();
+    component.handleClickOutside();
+  }
+
+  function changeResources(
+    original: ResourceCheckedSection[], changed: ResourceCheckedSection[], firstChange) {
+    // Programmatic change to an @Input does *not* trigger ngOnChanges like in the actual view,
+    // so have to do that explicitly
+    component.resources = changed;
+    component.ngOnChanges({ resources: new SimpleChange(original, changed, firstChange) });
+  }
+
+  function genResourceList(...resources: ResourceChecked[]): ResourceCheckedSection {
+    return {
+      title: 'any',
```

Comment:
Isn't the title optional? What do we gain from specifying one here?

Reply:
That causes testing failures: when comparing resources to filteredResources...
```
 expect(component.filteredResources).toEqual(resources);
```
Each item in `filteredResource` has a title set to `undefined`, but if we omit the title here on L380, the title is completely absent on each item in `resources`, so they do not match. Will add a comment.